### ISSUE TITLE
Unexplained scientists missing

### ISSUE THREAD

**Issue by geist22** [swfans/swars#19]
Issue from the original game.

Sometimes when you check your research tab after a mission, you'll find you have fewer scientists than before, despite the statistics screen not mentioning anything about scientist deaths. Is this a bug, or maybe they just ran away?

IMHO such an important change should always be mentioned in the statistics screen.

Using cheats to fast forward research while in the menu, you can see the number of your scientists dropping as time goes. Maybe this is a different mechanics unrelated to scientist deaths, because one mechanic for it wasn't annoying enough?

**Comment by geist22** [comment]
There's no need for additional display of screens - time doesn't advance outside of missions, so the only way for scientist numbers to drop is using the cheat to advance time/research. Otherwise, ALL scientists go missing during missions. It's okay for the numbers to go down without messages when using cheats. because it's not normal gameplay and you're on the research screen anyway when it happens so you see it right away.

It's just that like 4 out of 5 times, there's no mention of it on the statistics screen after a mission. You come back from a mission, check research and see you've lost scientists, but the game didn't tell you. One should not have to remember the number of their scientists and go check if they've lost any first thing after every mission. You need a way to force the "scientists died: number - reason" line on the statistics screen after a mission 100% of the time.

**Comment by mefistotelis** [comment]
>  time doesn't advance outside of missions

The function which shows time in the screen corner has a code to update research:

```
  if ( time_difference(&global_date, &research_curr_wep_date) )
  {
    if ( byte_1C497B )
      byte_1C497B = 0;
  }
  else if ( !byte_1C497B )
  {
    research_daily_progress_for_type(0);
    byte_1C497B = 1;
  }
```

Is it really inactive? Or does the time just pass slowly?

**Comment by mefistotelis** [comment]
I fixed one reason of such problems with this commit:
https://github.com/swfans/swars/commit/17518ee00c52f56a4471bc9215709f39559216ef#diff-94fb337d7329cc8d07d0a1478b275494a6b01dd63e7ed865a80cddd1f5305221L8732-R8835

Not sure it that's all.
Also, the issue would return if we fix the `research_curr_wep_date`.

**Comment by geist22** [comment]
I thought I saw my scientist number drop by one right at the start of my tests, but I can't seem to get it again.
Now I've completed about 200 or so missions with the Alt+C cheat and it didn't happen once. Maybe I was mistaken and missed something, or the issue could be connected to the seed?

I might do more tests later, but for now I can't reproduce this issue anymore, I think we can consider this fixed.
